src/System.Private.CoreLib/shared/System/IO/Path.cs

```diff
@@ -824,5 +824,35 @@ internal static StringComparison StringComparison
                     StringComparison.OrdinalIgnoreCase;
             }
         }
+
+        /// <summary>
+        /// Trims the ending directory separator if present.
+        /// </summary>
+        /// <param name="path"></param>
+        public static string TrimEndingDirectorySeparator(string path) =>
+            EndsInDirectorySeparator(path) && !PathInternal.IsRoot(path.AsSpan()) ?
+                path.Substring(0, path.Length - 1) :
+                path;
+
+        /// <summary>
+        /// Trims the ending directory separator if present.
+        /// </summary>
+        /// <param name="path"></param>
+        public static ReadOnlySpan<char> TrimEndingDirectorySeparator(ReadOnlySpan<char> path) =>
+            EndsInDirectorySeparator(path) && !PathInternal.IsRoot(path) ?
+                path.Slice(0, path.Length - 1) :
+                path;
+
+        /// <summary>
+        /// Returns true if the path ends in a directory separator.
+        /// </summary>
+        public static bool EndsInDirectorySeparator(ReadOnlySpan<char> path)
+            => path.Length > 0 && PathInternal.IsDirectorySeparator(path[path.Length - 1]);
+
+        /// <summary>
+        /// Returns true if the path ends in a directory separator.
+        /// </summary>
+        public static bool EndsInDirectorySeparator(string path)
+              => path != null && path.Length > 0 && PathInternal.IsDirectorySeparator(path[path.Length - 1]);
```

Comment:
@jkotas @stephentoub @JeremyKuhne unrelated question, opinions about `path is object` vs `!(path is null)` vs `path != null`?

Reply:
`path is object` obfuscates the intent. I would never use it for a null check.

`!(path is null)` vs `path != null`: 
- Prefer consistent style within file or component. `System.IO` is on `== null` plan, so it is what I would use here. 
- Prefer `is null` for types with overloaded `operator==`, like `System.Type` or delegates.
- When everything is equal, break the syntactic sugar ties by preferring one that has fewer characters. `path != null` has fewer characters than `!(path is null)`.